### ISSUE TITLE
Issue in first sync for documents with large history 

### ISSUE THREAD

**Issue by ashihaby** [couchbase/couchbase-lite-java-core#1317]
Hi, 
When I try to sync some documents from sync gateway server, the pull replicator fails to pull documents with high revisions count and when I check sync logs, I am not getting any errors but I see some transactions canceled like that.
``` 
07-01 00:57:28.521    3840-4228/com.xxx V/Sync﹕ com.couchbase.lite.replicator.PullerInternal$9@435d5ae0: inserting doc:db519257-26da-477f-818b-9e3de3bb43a7_f95c9e78-5f70-4f49-9f27-2a901ff6045c [23-1993723277d6f8b91891a94a6d5b9fb8, 22-901a1e9b672201d1bb0556bcdee7121b, 21-c61e5085c7df16995eeaf7bfc2ef2594, 20-ea76e55f53c5d5b4f3b95e85258cbe9c, 19-facb4cb3f30cd0eeeeefb32296789aef, 18-38d827949db31012c10c688e13609778, 17-acd5d18cffbc5b7e37103dea3c3e9a6a, 16-b36a7e9cfdc7232d9051b493dc10d24a, 15-9449367bb587a8f3b43aeee389d4a4d3, 14-57a4de3fff9ae0d56c08cef8201e21f6, 13-330cd20950b98b1c17c50d17edc02493, 12-1b3bb5ad2744477d0e46516ff0d422d9, 11-c807a9f3ebb09d2c6a6ed43632e5868a, 10-b6d9131b333efab81464446cb2099d5b, 9-563cc3bb3172ca8fb5b7620f97f3eeaf, 8-d17a30484b8c3ef1c5e78a2305b337c1, 7-8150fbe1a43b1eeb13f4cc031cbe8b5d, 6-e00c74fc90bad7627407111ef56a647e, 5-f2656d03fb176bf4405668235615dde0, 4-2ded370c5c7f4d6719635fc53b88d325, 3-92aa869f22d4fa1295f5cb9e737d5abd, 2-b2602918cde7b9b67253d73dba3a07af, 1-a918a0c48c1506497fda294fba24ecec]
07-01 00:57:28.521    3840-4228/com.xxx V/Database﹕ INSERT {doc:db519257-26da-477f-818b-9e3de3bb43a7_f95c9e78-5f70-4f49-9f27-2a901ff6045c #23-1993723277d6f8b91891a94a6d5b9fb8 @36}, history[23]
07-01 00:57:28.546    3840-4228/com.xxx V/Database﹕ CBLReplicationExecutor-https://rc-sync.xxx.com/xxx-pull-59c9d Begin transaction (level 1)
07-01 00:57:28.581    3840-4228/com.xxx V/Database﹕ CBLReplicationExecutor-https://rc-sync.xxx.com/xxx-pull-59c9d CANCEL transaction (level 1)
```
The issue exists on the master branch, I have tried version 1.1.0 and the issue didn't appear in old versions of couchbase lite.

Thanks

**Comment by hideki** [comment]
Which version of CBL are you using?

**Comment by ashihaby** [comment]
@hideki the issue exist on master branch and on this version `0.0.0-755`

**Comment by hideki** [comment]
Hi @ashihaby,

Thank you for your additional comments. I updated my previous comments. Could you please provide more information?

Can  you write simple sample app, or can you provide sync gateway endpoint that we can try ?

**Comment by ashihaby** [comment]
after debugging I found NullPointerException when I pull documents from sync_gateway into empty database. Please check the fix here #1318

**Comment by hideki** [comment]
Hi @ashihaby,
Thank you very much for your PR. I think this fix just null check. In general, could you please write test or please provide steps to reproduce the issue? We could not blindly accept PRs.
Thanks again!
